src/match.js

```diff
@@ -0,0 +1,118 @@
+const isStar = (char) => char === '*';
+
+const isPlus = (char) => char === '+';
+
+const isQuestion = (char) => char === '?';
+
+const isOperator = (char) => isStar(char) || isPlus(char) || isQuestion(char);
+
+const isLiteral = (ch) => (ch >= "a" && ch <= "z") || (ch >= "A" && ch <= "Z") || (ch >= "0" && ch <= "9");
+
+const isSet = (term) => isOpenSet(term[0]) && isCloseSet(term.slice(-1));
+
+const isUnit = (term) => isLiteral(term[0]) || isSet(term)
+
+const isOpenSet = (char) => char === '[';
+
+const isCloseSet = (char) => char === ']';
+
+function removeSlashWrapper(expr) {
+  if (expr[0] === '/' && expr.slice(-1) === '/') {
+    return expr.slice(1, -1);
+  } else {
+    throw new Error('Incorrect RegExp. The RegExp must be wrapped in slashes');
+  }
+}
+
+function splitSet(set_head) {
+  const set_inside = set_head.slice(1, -1);
+  const set_terms = set_inside.split('');
+  return set_terms;
+}
```

Comment:
You can write this function in one line. Then it will be faster to understand what it is doing
```suggestion
function splitSet(set_head) {
  return set_head.slice(1, -1).split('');
}
```

Reply:
Thanks for the review. The code will be refactored